test/unit-test/mqtt_agent_utest.c

```diff
@@ -1334,6 +1341,53 @@ void test_MQTTAgent_CommandLoop_with_eventCallback( void )
     /* Ensure that callback is invoked. */
     TEST_ASSERT_EQUAL( 5, commandCompleteCallbackCount );
 
+    /* mqttEventcallback behavior when the packet Id 0 is received in the PUBACK. */
+    packetIdentifier = 0U;
+    mqttAgentContext.pPendingAcks[ 0 ].packetId = 0U;
+    command.pCommandCompleteCallback = stubCompletionCallback;
+    mqttAgentContext.pPendingAcks[ 0 ].pOriginalCommand = &command;
+
+    MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTT_ProcessLoop_StubWithCallback( MQTT_ProcessLoop_stub );
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that callback is invoked. */
+    TEST_ASSERT_EQUAL( 6, commandCompleteCallbackCount );
+
+    /* Test case checking no corresponding ack exists for the PUBACK received in the pPendingAcks array. */
```

Comment:
I think `packetIdentifier` would need be reset before this, right?

Reply:
Yeah Resolved it